pkg/querier/distributor_queryable.go

```diff
@@ -54,6 +70,45 @@ func (q *distributorQuerier) Select(sp *storage.SelectParams, matchers ...*label
 	return matrixToSeriesSet(matrix), nil, nil
 }
 
+func (q *distributorQuerier) streamingSelect(sp storage.SelectParams, matchers []*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
+	userID, err := user.ExtractOrgID(q.ctx)
+	if err != nil {
+		return nil, nil, promql.ErrStorage{Err: err}
+	}
+
+	mint, maxt := sp.Start, sp.End
+
+	results, err := q.distributor.QueryStream(q.ctx, model.Time(mint), model.Time(maxt), matchers...)
+	if err != nil {
+		return nil, nil, promql.ErrStorage{Err: err}
+	}
+
+	serieses := make([]storage.Series, 0, len(results))
+	for _, result := range results {
+		// Sometimes the ingester can send series that have no data.
+		if len(result.Chunks) == 0 {
+			continue
+		}
+
+		ls := client.FromLabelAdaptersToLabels(result.Labels)
+		sort.Sort(ls)
+
+		chunks, err := chunkcompat.FromChunks(userID, ls, result.Chunks)
```

Comment:
Before we were passing `nil` for the labels. Is this change intended?

Reply:
Yes, it is intended.

Code you're referring to was in `ingesterStreamingQuerier.Select` method before, and it is correct that we didn't set labels into the chunk there. However, `ingesterStreamingQuerier` was only ever used in combination with `unifiedChunkQuerier`, which does not call its `Select` method, but `Get` method instead. I believe that `Select` was only ever used from `TestIngesterStreaming`.

Reason for setting labels into chunk directly is so that `partitionChunks` can sort chunks into individual series correctly (via fingerprinting). It relies on this field to be set.